testing/trino-tests/src/test/java/io/trino/connector/system/runtime/TestKillQuery.java

```diff
@@ -69,6 +69,13 @@ public void tearDown()
     @Test(timeOut = 60_000)
     public void testKillQuery()
             throws Exception
+    {
+        killQuery(Optional.of("because"), "Message: because");
+        killQuery(Optional.empty(), "No message provided.");
+    }
+
+    private void killQuery(Optional<String> argumentMessage, String expectedKilledMessage)
+            throws Exception
```

Comment:
Avoid throwing `Exception`. Throwing `Exception` is fine in test methods, but this is a helper method, not a test method. 

Another approach is using `Uninterruptibles.sleepUninterruptibly`.